maven-model-builder/src/main/java/org/apache/maven/model/building/FileModelSource.java

```diff
@@ -80,4 +80,27 @@ public URI getLocationURI()
     {
         return getFile().toURI();
     }
+
+    @Override
+    public boolean equals( Object obj ) 
+    {
+        if ( this == obj ) 
+        {
+            return true;
+        }
+        
+        if ( ( obj == null ) || !( obj instanceof FileModelSource ) ) 
```

Comment:
Compare with null not necessary since `null instanceof FileModelSource` is `false`

Reply:
You´re right this make code simpler, thank you